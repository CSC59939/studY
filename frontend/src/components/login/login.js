import React from 'react';
import { TextField } from '@material-ui/core';
import classes from './login.module.css';
import icon from '../../images/icon.png'
// more components at https://material-ui.com/getting-started/usage/
import { withFormik, Field, Form } from 'formik';
import * as Yup from 'yup';
import { loginUser } from '../../fetches';

const inputField = ({ 
	input, children, id, 
	label, type, placeholder,
	required, onChange, helperText
}) => (
	<TextField InputProps={{className: classes.TextField}}
		id={id} label={label} type={type}
		variant='outlined' {...input}
		placeholder={placeholder} margin='dense'
		children={children} required={required}
		fullWidth onBlur={onChange} helperText={helperText} 
	/>
)

export const Login = props => {
		const { handleChange, errors, touched, isSubmitting } = props;

		return (
			<div className={classes.Container}>
				<div className={classes.PageColumns}>
					<div className={classes.PageColumn_left}>
						<div className={classes.Art}>
						</div>
					</div>
					<div className={classes.PageColumn_right}>
						<div className={classes.ColumnContainer}>
							<div>
								<img alt="icon" src={icon} className={classes.Icon}/>    
							</div>

							<h1 className={classes.Title}>Sign In</h1>
							<Form name='loginForm' id="myForm" className={classes.Form} >
								<Field 
								name='email' 
								id='email'
								label='Email' 
								type='email' 
								helperText={ touched.email && errors.email && <p>{errors.email}</p> }
								onChange={handleChange}
								component={inputField} />

								<Field 
								name='password' 
								id='password'
								label='Password' 
								type='password' 
								helperText={touched.password && errors.password && <p>{errors.password}</p>}
								onChange={handleChange}
								component={inputField} />
								{ errors.loginForm && <span>{errors.loginForm}</span> }

								<div>
									<button type="submit" disabled={isSubmitting} className={classes.Submit}>
										SIGN IN
									</button>
								</div>
							</Form>
						</div>
					</div>
				</div>
			</div>
		)
	}

export default withFormik({
	mapPropsToValues: () => ({
		email: '',
		password: '',
	}),
	validationSchema: Yup.object().shape({
		email: Yup.string()
			.matches(/(.cuny.edu)/, 'Must be a valid CUNY Email')
			.required('Email is required'),
		password: Yup.string().required(),
	}),
  handleSubmit: (user, { props, setErrors, setSubmitting }) => {
    loginUser(user).then(result => {
      props.onLogin(result.token)
    })// .then(() => props.history.push('/'))
    .catch(error => {
			setErrors({ loginForm: error.message});
			setSubmitting(false);
		})
	}
})(Login);
