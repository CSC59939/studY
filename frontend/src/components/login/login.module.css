.Form {
    background-color: #4b58a7;
    width: 35%;
    margin: 130px auto;
    text-align: center;
    font-size: normal;
    box-shadow: 0px 0px 8px 0px rgba(10, 10, 10, 0.75);
    display: flex;
    flex-flow: column;
}

.Login {
    height: 70px;
    background-color: #282e78;
    color: white;
    font-size: x-large;
    letter-spacing: 1.5px;
    vertical-align: middle;
}

.TextField {
    background-color: white;
    width: 85%;
    height: 50px;
}

.Button {
    background-color: #969cce;
    width: 85%;
    height: 45px;
    margin: 20px auto; 
    border: none;
    color: white;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 20px;
    cursor:pointer;
    overflow: hidden;
}

.RegisterButton {
    background-color: Transparent;
    background-repeat:no-repeat;
    border: none;
    cursor:pointer;
    overflow: hidden;
    outline:none;
    font-size: middle;
}

.Outer {
    padding: 30px 10px;
}


@media (min-width: 300px) and (min-height: 200px) {
    .Form {
        width: 300px;
    }
}

@media (min-width: 400px) and (min-height: 301px) {
    .Form {
        width: 400px;
    }
}

@media (min-width: 1000px) and (min-height: 700px) {
    .Form {
        width: 35%;
    }
}