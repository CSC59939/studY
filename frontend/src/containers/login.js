
import React, { Component } from 'react';

import { Login } from '../components';
import { AuthContext, AuthWrapper } from '../contexts/Auth.context';
import { withFormik } from 'formik';

class loginWrapper extends Component {
  static contextType = AuthContext;

  render() {
    return (
      <Login {...this.props} {...this.context} />
    )
  }
}

export default (loginWrapper);