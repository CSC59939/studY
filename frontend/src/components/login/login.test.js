import React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

import { Login } from './login';

enzyme.configure({ adapter: new Adapter() });

describe('Login Component', () => {
	it('should render without throwing an error', () => {
		expect(shallow(<Login />).find('#myForm').exists()).toBe(true)
	})
})

it('renders a email input', () => {
	expect(shallow(<Login />).find('#outlined-email-input').length).toEqual(1)
})

it('renders a password input', () => {
	expect(shallow(<Login />).find('#outlined-password-input').length).toEqual(1)
})

describe('Email input', () => {
  
	it('should respond to change event and change the state of the Login Component', () => {
	 
	const wrapper = shallow(<Login />);
	wrapper.find('#outlined-email-input').simulate('change', {target: {name: 'email', value: 'blah@gmail.com'}});
	 
	expect(wrapper.state().user.undefined).toEqual('blah@gmail.com');
	})
})
   
describe('Password input', () => {
	
	it('should respond to change event and change the state of the Login Component', () => {
	 
	const wrapper = shallow(<Login />);
   	wrapper.find('#outlined-password-input').simulate('change', {target: {name: 'password', value: '1234'}});
   
    expect(wrapper.state().user.undefined).toEqual('1234');
  	})
})




