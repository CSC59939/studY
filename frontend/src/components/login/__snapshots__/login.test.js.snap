// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component snapshot 1`] = `
<div
  className="Container"
>
  <div
    className="PageColumns"
  >
    <div
      className="PageColumn_left"
    >
      <div
        className="Art"
      />
    </div>
    <div
      className="PageColumn_right"
    >
      <div
        className="ColumnContainer"
      >
        <div>
          <img
            alt="icon"
            className="Icon"
            src="icon.png"
          />
        </div>
        <h1
          className="Title"
        >
          Sign In
        </h1>
        <form
          className="Form"
          id="myForm"
          name="loginForm"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root-1 MuiFormControl-marginDense-3 MuiFormControl-fullWidth-4"
            field={
              Object {
                "name": "email",
                "onBlur": [Function],
                "onChange": [Function],
                "value": "",
              }
            }
            form={
              Object {
                "dirty": false,
                "errors": Object {},
                "handleBlur": [Function],
                "handleChange": [Function],
                "handleReset": [Function],
                "handleSubmit": [Function],
                "initialValues": Object {
                  "email": "",
                  "password": "",
                },
                "isSubmitting": false,
                "isValid": false,
                "isValidating": false,
                "registerField": [Function],
                "resetForm": [Function],
                "setError": [Function],
                "setErrors": [Function],
                "setFieldError": [Function],
                "setFieldTouched": [Function],
                "setFieldValue": [Function],
                "setFormikState": [Function],
                "setStatus": [Function],
                "setSubmitting": [Function],
                "setTouched": [Function],
                "setValues": [Function],
                "submitCount": 0,
                "submitForm": [Function],
                "touched": Object {},
                "unregisterField": [Function],
                "validateField": [Function],
                "validateForm": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "values": Object {
                  "email": "",
                  "password": "",
                },
              }
            }
          >
            <label
              className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-marginDense-11 MuiInputLabel-outlined-15"
              data-shrink={false}
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="MuiInputBase-root-36 MuiOutlinedInput-root-23 MuiInputBase-fullWidth-45 MuiInputBase-formControl-37 MuiInputBase-marginDense-43 TextField"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="MuiPrivateNotchedOutline-legend-54"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input-46 MuiOutlinedInput-input-31 MuiInputBase-inputType-49 MuiInputBase-inputMarginDense-47 MuiOutlinedInput-inputMarginDense-32"
                disabled={false}
                id="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="email"
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root-1 MuiFormControl-marginDense-3 MuiFormControl-fullWidth-4"
            field={
              Object {
                "name": "password",
                "onBlur": [Function],
                "onChange": [Function],
                "value": "",
              }
            }
            form={
              Object {
                "dirty": false,
                "errors": Object {},
                "handleBlur": [Function],
                "handleChange": [Function],
                "handleReset": [Function],
                "handleSubmit": [Function],
                "initialValues": Object {
                  "email": "",
                  "password": "",
                },
                "isSubmitting": false,
                "isValid": false,
                "isValidating": false,
                "registerField": [Function],
                "resetForm": [Function],
                "setError": [Function],
                "setErrors": [Function],
                "setFieldError": [Function],
                "setFieldTouched": [Function],
                "setFieldValue": [Function],
                "setFormikState": [Function],
                "setStatus": [Function],
                "setSubmitting": [Function],
                "setTouched": [Function],
                "setValues": [Function],
                "submitCount": 0,
                "submitForm": [Function],
                "touched": Object {},
                "unregisterField": [Function],
                "validateField": [Function],
                "validateForm": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "values": Object {
                  "email": "",
                  "password": "",
                },
              }
            }
          >
            <label
              className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-marginDense-11 MuiInputLabel-outlined-15"
              data-shrink={false}
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="MuiInputBase-root-36 MuiOutlinedInput-root-23 MuiInputBase-fullWidth-45 MuiInputBase-formControl-37 MuiInputBase-marginDense-43 TextField"
              onClick={[Function]}
            >
              <fieldset
                aria-hidden={true}
                className="MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="MuiPrivateNotchedOutline-legend-54"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
              <input
                aria-invalid={false}
                className="MuiInputBase-input-46 MuiOutlinedInput-input-31 MuiInputBase-inputType-49 MuiInputBase-inputMarginDense-47 MuiOutlinedInput-inputMarginDense-32"
                disabled={false}
                id="password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
            </div>
          </div>
          <div>
            <button
              className="Submit"
              disabled={false}
              type="submit"
            >
              SIGN IN
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
